src/groups.js

```diff
@@ -0,0 +1,194 @@
+const assert = require("assert").strict;
+const {NotFound} = require("http-errors");
+
+const GROUPS_DATA = require("../data/groups.json");
+
+const PRODUCTION = process.env.NODE_ENV === "production";
+
+/**
+ * Map of Nextstrain Groups from their (normalized) name to their static config
+ * record.
+ *
+ * This essentially takes the place of what might be a database table in the
+ * future but is perfectly fine as a flat file key-value store for now.
+ */
+const GROUP_RECORDS = new Map(
+  GROUPS_DATA
+    /* Groups for dev/testing.  Might be nice to expose these in production too,
+     * but we'd need additional changes first to support "unlisted" Groups.
+     *   -trs, 24 Jan 2022
+     * */
+    .filter(groupRecord => PRODUCTION ? !groupRecord.isDevOnly : true)
+    .map(groupRecord => [normalizeGroupName(groupRecord.name), groupRecord])
+);
+
+assert(GROUP_RECORDS.size === GROUPS_DATA.length);
```

Comment:
Would this assertion fail in production since we filter out the `isDevOnly` groups?

Reply:
Oh, it sure would! :grimacing: Nice catch! I'll fix that.